tuf/signed/verify.go

```diff
@@ -92,7 +93,7 @@ func VerifySignatures(s *data.Signed, roleData data.BaseRole) error {
 }
 
 // VerifySignature checks a single signature and public key against a payload
```

Comment:
nit: we should mention that this function will actually mutate the signature to mark it as valid if it verifies against the public key and message in the description